File: client/src/types/models/permission.ts
import { BaseModel } from "./base";
import { Role } from "./role";

export interface Permission extends BaseModel {
  id: string;
  name: string;
  description: string;
  read: boolean;
  readDescription: string;
  update: boolean;
  updateDescription: string;
  create: boolean;
  createDescription: string;
  roleId: string;

  role?: Role;
}

File: core/internal/models/permission.go
package models

import "gorm.io/gorm"

type Permission struct {
	ID                string `gorm:"type:uuid;default:uuid_generate_v4();primaryKey"`
	Name              string `gorm:"type:varchar(255);not null"`
	Description       string `gorm:"type:text"`
	Read              bool   `gorm:"default:false"`
	ReadDescription   string `gorm:"type:text"`
	Update            bool   `gorm:"default:false"`
	UpdateDescription string `gorm:"type:text"`
	Create            bool   `gorm:"default:false"`
	CreateDescription string `gorm:"type:text"`

	// Foreign key
	RoleID string `gorm:"type:uuid;not null"`
	Role   Role   `gorm:"foreignKey:RoleID"`
	gorm.Model
}
